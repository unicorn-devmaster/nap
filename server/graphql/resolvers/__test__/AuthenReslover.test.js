/* eslint-env jest */
describe('AuthenResolver', () => {
  it('should authen and return user', async () => {
    // stub
    global.NAP = {}
    NAP.Authen = {
      findOneAndUpdate: jest.fn().mockImplementationOnce(() =>
        Promise.resolve({ _id: '58d0e20e7ff032b39c2a9a18', name: 'bar' })
      )
    }

    const authen = require('../AuthenResolver')
    const installationId = 'FOO_INSTALLATION_ID'
    const userObject = { id: 'FOO_USER_ID' }
    const provider = 'facebook'

    const user = await authen.willAuthen(installationId, userObject, provider)

    expect(user).toMatchSnapshot()
  })
})