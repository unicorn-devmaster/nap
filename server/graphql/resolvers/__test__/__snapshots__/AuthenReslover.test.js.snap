// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenResolver should authen and return user 1`] = `
Object {
  "_id": "58d0e20e7ff032b39c2a9a18",
  "name": "bar",
}
`;
